### PR TITLE
Bug 1187314 - clock - timer does not remember last settings.

### PR DESCRIPTION
**Pull request by mcav**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=da04fb8808b00b93ab1aa66bf31696aa0afdecde

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=545d82c438c2b8141436ccd61863709f2b2bb3f9

File: apps/clock/js/panels/timer/main.js [hunk 1]
```diff
@@ -109,18 +110,25 @@ Timer.Panel = function(element) {
...
 
   // The start button is enable if the value of the timer is not 00:00
   picker.nodes.minutes.addEventListener('transitionend', enableButton);
   picker.nodes.hours.addEventListener('transitionend', enableButton);
 
+  this.restoreDefaults();
+  this.nodes.vibrate.addEventListener(
+    'change', (e) => this.saveCurrentSettingsAsDefaults());
+  this.nodes.sound.addEventListener(
+    'change', (e) => this.saveCurrentSettingsAsDefaults());
```
Comment (zbraniecki) [inline apps/clock/js/panels/timer/main.js]: I'm surprised your pre-commit linter doesn't yell about unused variable `e`.

File: apps/clock/js/panels/timer/main.js [hunk 2]
```diff
@@ -140,6 +148,29 @@ Timer.Panel = function(element) {
 
 Timer.Panel.prototype = Object.create(Panel.prototype);
 
+Timer.Panel.prototype.restoreDefaults = function() {
+  asyncStorage.getItem('timer-defaults', (values) => {
+    values = values || {};
+    if (values.sound !== null) {
```
Comment (zbraniecki) [inline apps/clock/js/panels/timer/main.js]: you may want to check for both `null` and `undefined` here.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a04f7d82a66e956d7a0c46dabe6498e69c588f1c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a04f7d82a66e956d7a0c46dabe6498e69c588f1c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ef465338ac4a65abec604e60f24f757d48ebdc2e

**Pull request merged**